docs/HowToFAQ/FAQTriggers.md

```diff
@@ -0,0 +1,22 @@
+---
+title: "What does `forall v :: v in vals ==> false` evaluate to if `vals` is non-empty?"
+---
+
+## Question
+
+What does `forall v :: v in vals ==> false` evaluate to if `vals` is non-empty?
+Should it be false? I’m having some problem proving the last assertion in Dafny.
+
+```dafny
+assert vals != [];
+assert MatchingResult() == (forall v :: v in vals ==> false);
+assert !MatchingResult();
+```
+
+## Answer
+
+The problem here is the trigger on the quantification expression.
+Dafny sees the `forall` quantifier and but uses it only when there is a `v in vals` fact in the context for some v (this is what the annotation on the forall iin the VSCode IDE means: Selected triggers: {v in vals}).  So until you give it a concrete fact to "trigger" on, it doesn't use your quantifier.
```

Comment:
"So until you give it a concrete fact to "trigger" on, it doesn't use your quantifier." shouldn't we show how to do that for this example? Linking to a paper for the final answer for more information seems fine, but as the final answer to this simple question it seems heavy handed.

Reply:
I'll make a note to find and add an example.

Reply:
Added a reference and txt after discussion with Rustan.